OSDOCS-9882-manage-console: Bug batched the 4.16 management console bugs (#77706)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

[id="ocp-4-16-management-console-bug-fixes_{context}"]
==== Management Console

* Previously, the *Debug container* link was not shown for pods with a `Completed` status. With this release, the link shows as expected. (link:https://issues.redhat.com/browse/OCPBUGS-34711[*OCPBUGS-34711*])

> * Previously, select `<textarea>` were not resizable due to Patternfly 5 requiring `pf-m-resize-both`.  With this release, `pf-m-resize-both` was added and `<textarea>` is now resizable. (link:https://issues.redhat.com/browse/OCPBUGS-34393[*OCPBUGS-34393*])

Reviewer: ```suggestion
* Previously, due to an issue in PatternFly 5, text boxes in the {product-title} web console were no longer resizable.  With this release, text boxes are again resizable. (link:https://issues.redhat.com/browse/OCPBUGS-34393[*OCPBUGS-34393*])
```

* Previously, French and Spanish were not available in the web console. With this release, translations for French and Spanish are now available. (link:https://issues.redhat.com/browse/OCPBUGS-33965[*OCPBUGS-33965*])

> * Previously, the masthead logo was not restricted to a max-height of 60px, and logos larger that 60px making the masthead size too large. With this release, the masthead logo is restricted to a max-height of 60px. (link:https://issues.redhat.com/browse/OCPBUGS-33523[*OCPBUGS-33523*])

Reviewer: I got this wording from the bug description
```suggestion
* Previously, the masthead logo was not restricted to a max-height of 60px. As a result, logos that are larger than 60px high display at their native size and cause the masthead to be too large. With this release, the masthead logo is restricted to a max-height of 60px. (link:https://issues.redhat.com/browse/OCPBUGS-33523[*OCPBUGS-33523*])
```

* Previously, there was a missing return statement in the `HealthCheck` controller causing it to panic under certain circumstances. With this release, the proper return statement was added to the `HealthCheck` controller so it no longer panics. (link:https://issues.redhat.com/browse/OCPBUGS-33505[*OCPBUGS-33505*]

> * Previously, an incorrect field was sent to the API server which was not noticeable, but with the implementation of Admission Webhook display warning the same action would return a warning notification. A fix was provided to resolve the issue. (link:https://issues.redhat.com/browse/OCPBUGS-33222[*OCPBUGS-33222*])

Reviewer: ```suggestion
* Previously, an incorrect field was sent to the API server that was not noticeable. With the implementation of Admission Webhook display warning the same action would return a warning notification. A fix was provided to resolve the issue. (link:https://issues.redhat.com/browse/OCPBUGS-33222[*OCPBUGS-33222*])
```

* Previously, the message text of a `StatusItem` might have been vertically misaligned with the icon when a timestamp was not present. With this release, the message text is correctly aligned. (link:https://issues.redhat.com/browse/OCPBUGS-33219[*OCPBUGS-33219*])

> * Previously, the creator field was autopopulated and not mandatory. Updates to the API made the field empty from 4.15 on. With this release, the field is marked as mandatory for correct validation. (link:https://issues.redhat.com/browse/OCPBUGS-31931[*OCPBUGS-31931*])

Reviewer: ```suggestion
* Previously, the creator field was autopopulated and not mandatory. Updates to the API made the field empty from {product-title} 4.15 and higher. With this release, the field is marked as mandatory for correct validation. (link:https://issues.redhat.com/browse/OCPBUGS-31931[*OCPBUGS-31931*])
```

> * Previously, the `ResourceYAMLEditor` was always used in the resource details page to edit objects, but when it was used to create objects no *Create* button or samples appeared. With this release, additional parameters were added so that you can create objects by using the `ResourceYAMLEditor`. You can now see the *Create* button and samples. (link:https://issues.redhat.com/browse/OCPBUGS-31703[*OCPBUGS-31703*])

Reviewer: ```suggestion
* Previously, the YAML editor in the web console did not have a *Create* button and samples did not appear. With this release, you can now see the *Create* button and samples. (link:https://issues.redhat.com/browse/OCPBUGS-31703[*OCPBUGS-31703*])
```

* Previously, changes to the bridge server flags on an external OpenID Connect (OIDC) feature caused the bridge server fail to start in local development. With this release, the flags usage are updated and the bridge server starts. (link:https://issues.redhat.com/browse/OCPBUGS-31695[*OCPBUGS-31695*])

> * Previously, when editing a {vmw-full} connection, the form could be submitted even if no values were actually changed. This lead to the nodes rebooting. With this release, the console now detects the form changes, and does not allow submission if no value was changed. (link:https://issues.redhat.com/browse/OCPBUGS-31613[*OCPBUGS-31613*])

Reviewer: ```suggestion
* Previously, when editing a {vmw-full} connection, the form could be submitted even if no values were actually changed. This resulted in the nodes rebooting. With this release, the console now detects the form changes, and does not allow submission if no value was changed. (link:https://issues.redhat.com/browse/OCPBUGS-31613[*OCPBUGS-31613*])
```

* Previously, plugins using `ListPageFilters` were only using two filters: label and name. With this release, a parameter was added that enables plugins to configure multiple text-based search filters. (link:https://issues.redhat.com/browse/OCPBUGS-30077[*OCPBUGS-30077*])

* Previously, there was no response when clicking on quick start items. With this release, the quick start window shows when clicking on the quick start selections. (link:https://issues.redhat.com/browse/OCPBUGS-29992[*OCPBUGS-29992*])

* Previously, the {product-title} web console terminated unexpectedly if authentication discovery failed on the first attempt. With this release, authentication initialization was updated to retry up to 5 minutes before failing. (link:https://issues.redhat.com/browse/OCPBUGS-29479[*OCPBUGS-29479*])

* Previously there was an issue causing an error message on the *Image Manifest Vulnerability* page after an Image Manifest Vulnerability (IMV) was created in the CLI. With this release, the error message no longer shows. (link:https://issues.redhat.com/browse/OCPBUGS-28967[*OCPBUGS-28967*])

> * Previously, when using the modal dialog in a hook as part of the actions hook, an error occurs because the console framework will pass null objects as part of the render cycle. With this release, `getGroupVersionKindForResource` is now null safe and will return `undefined` if the `apiVersion` or `kind` are undefined. Additionally, the run time error for `useDeleteModal` no longer occurs, but note that it will not work with an `undefined` resource. (link:https://issues.redhat.com/browse/OCPBUGS-28856[*OCPBUGS-28856*])

Reviewer: ```suggestion
* Previously, when using the modal dialog in a hook as part of the actions hook, an error occurred because the console framework passed null objects as part of the render cycle. With this release, `getGroupVersionKindForResource` is now null-safe and will return `undefined` if the `apiVersion` or `kind` are undefined. Additionally, the run time error for `useDeleteModal` no longer occurs, but note that it will not work with an `undefined` resource. (link:https://issues.redhat.com/browse/OCPBUGS-28856[*OCPBUGS-28856*])
```

* Previously, the *Expand PersistentVolumeClaim* modal assumes the `pvc.spec.resources.requests.stroage` value includes a unit. With this release, the size is updated to 2GiB and you can change the value of the persistent volume claim (PVC). (link:https://issues.redhat.com/browse/OCPBUGS-27779[*OCPBUGS-27779*])

* Previously, the value of image vulnerabilities reported in the {product-title} web console were inconsistent. With this release, the image vulnerabilities on the *Overview* page were removed. (link:https://issues.redhat.com/browse/OCPBUGS-27455[*OCPBUGS-27455*])

> * Previously, a certificate signing request (CSR) might appear for a recently approved Node. With this release, the duplication is detected and does not show CSRs for approved Nodes. (link:https://issues.redhat.com/browse/OCPBUGS-27399[*OCPBUGS-27399*])

Reviewer: ```suggestion
* Previously, a certificate signing request (CSR) could appear for a recently approved Node. With this release, the duplication is detected and does not show CSRs for approved Nodes. (link:https://issues.redhat.com/browse/OCPBUGS-27399[*OCPBUGS-27399*])
```

* Previously, the *Type* column was not first on the condition table on the *MachineHealthCheck detail* page. With this release, the *Type* is now listed first on the condition table. (link:https://issues.redhat.com/browse/OCPBUGS-27246[*OCPBUGS-27246*])

* Previously, the console plugin proxy was not copying the status code from plugin service responses. This caused all responses from the plugin service to have a `200` status, causing unexpected behavior, especially around browser caching. With this release, the console proxy logic was updated to forward the plugin service proxy response status code. Proxied plugin requests now behave as expected. (link:https://issues.redhat.com/browse/OCPBUGS-26933[*OCPBUGS-26933*])

> * Previously, when cloning a persistent volume claim (PVC), the modal assumes pvc.spec.resources.requests.storage value includes a unit. With this release, `pvc.spec.resources.requests.storage` includes a unit suffix and the *Clone PVC* modal works as expected. (link: https://issues.redhat.com/browse/OCPBUGS-26772[*OCPBUGS-26772*])

Reviewer: ```suggestion
* Previously, when cloning a persistent volume claim (PVC), the modal assumes `pvc.spec.resources.requests.storage` value includes a unit. With this release, `pvc.spec.resources.requests.storage` includes a unit suffix and the *Clone PVC* modal works as expected. (link: https://issues.redhat.com/browse/OCPBUGS-26772[*OCPBUGS-26772*])
```

* Previously, escaped strings were not handled properly when editing {vmw-full} connection, causing broken {vmw-full} configuration. With this release, the escape strings work as expected and the {vmw-full} configuration no longer breaks. (link:https://issues.redhat.com/browse/OCPBUGS-25942[*OCPBUGS-25942*])

* Previously, when configuring a {vmw-full} connection,  the `resourcepool-path` key was not added to the {vmw-full} config map which might have caused issues connecting to {vmw-full}. With this release, there are no longer issues connecting to {vmw-full}. (link:https://issues.redhat.com/browse/OCPBUGS-25927[*OCPBUGS-25927*])

* Previously, there was missing text in the *Customer feedback* modal. With this release, the link text is restored and the correct Red Hat image is displayed. (link:https://issues.redhat.com/browse/OCPBUGS-25843[*OCPBUGS-25843*])

* Previously, the *Update cluster* modal would not open when clicking *Select a version* from the *Cluster Settings* page. With this release, the *Update cluster* modal shows when clicking *Select a version*. (link:https://issues.redhat.com/browse/OCPBUGS-25780[*OCPBUGS-25780*])

> * Previously, the filter part in the resource section of the *Search* page did not work on a mobile device. With this release, filtering now works as expected on a mobile device. (link:https://issues.redhat.com/browse/OCPBUGS-25530[*OCPBUGS-25530*])

Reviewer: ```suggestion
* Previously, on a mobile device, the filter part in the resource section of the *Search* page did not work. With this release, filtering now works as expected on a mobile device. (link:https://issues.redhat.com/browse/OCPBUGS-25530[*OCPBUGS-25530*])
```

> * Previously, the console Operator was using client instead of listeners for fetching a cluster resource causing the Operator to do operations on resources with old revision. With this release, the console Operator uses list to fetch data from cluster instead of clients. (link:https://issues.redhat.com/browse/OCPBUGS-25484[*OCPBUGS-25484*])

Reviewer: ```suggestion
* Previously, the console Operator was using a client instead of listeners for fetching a cluster resource. This caused the Operator to do operations on resources with an older revision. With this release, the console Operator uses list to fetch data from cluster instead of clients. (link:https://issues.redhat.com/browse/OCPBUGS-25484[*OCPBUGS-25484*])
```

* Previously, the console was incorrectly parsing restore size values from volume snapshots in the restore as new persistent volume claims  (PVC) modal. With this release, the modal parses the restore size correctly. (link:https://issues.redhat.com/browse/OCPBUGS-24637[*OCPBUGS-24637*])

> * Previously, *Alerting*, *Metrics* and *Target* pages were not available in the console due to a change on the routing library. With this release, routes load correctly. (link:https://issues.redhat.com/browse/OCPBUGS-24515[*OCPBUGS-24515*])

Reviewer: ```suggestion
* Previously, the *Alerting*, *Metrics*, and *Target* pages were not available in the console due to a change on the routing library. With this release, routes load correctly. (link:https://issues.redhat.com/browse/OCPBUGS-24515[*OCPBUGS-24515*])
```

> * Previously, there was a run time error on the *Node details* page when `MachineHealthCheck` without conditions exists. With this release, the *Node details* page loads as expected. (link:https://issues.redhat.com/browse/OCPBUGS-24408[*OCPBUGS-24408*])

Reviewer: ```suggestion
* Previously, there was a runtime error on the *Node details* page when a `MachineHealthCheck` without conditions existed. With this release, the *Node details* page loads as expected. (link:https://issues.redhat.com/browse/OCPBUGS-24408[*OCPBUGS-24408*])
```

* Previously, the console backend would proxy operand list requests to the public API server endpoint, which caused CA certificate issues under some circumstances. With this release, the proxy configuration was updated to point to the internal API server endpoint which fixed this issue. (link:https://issues.redhat.com/browse/OCPBUGS-22487[*OCPBUGS-22487*])

> * Previously, a deployment could not be scaled up or down when a `HorizontalPodAutoscaler` is present. With this release, when a deployment with an `HorizontalPodAutoscaler` is scaled down to `zero`, an *Enable Autoscale* button is displayed so you can enable pod autoscaling. (link:https://issues.redhat.com/browse/OCPBUGS-22405[*OCPBUGS-22405*])

Reviewer: ```suggestion
* Previously, a deployment could not be scaled up or down when a `HorizontalPodAutoscaler` was present. With this release, when a deployment with an `HorizontalPodAutoscaler` is scaled down to `zero`, an *Enable Autoscale* button is displayed so you can enable pod autoscaling. (link:https://issues.redhat.com/browse/OCPBUGS-22405[*OCPBUGS-22405*])
```

> * Previously, when editing a file an error `Info alert:Non-printable file detected. File contains non-printable characters. Preview is not available.` is presented. With this release, a check was added to determine if a file is binary, and you are able to edit the file as expected. (link:https://issues.redhat.com/browse/OCPBUGS-18699[*OCPBUGS-18699*])

Reviewer: ```suggestion
* Previously, when editing a file, the `Info alert:Non-printable file detected. File contains non-printable characters. Preview is not available.` error was presented. With this release, a check was added to determine if a file is binary, and you are able to edit the file as expected. (link:https://issues.redhat.com/browse/OCPBUGS-18699[*OCPBUGS-18699*])
```

* Previously, the console API conversion webhook server could not update serving certificates at runtime, and would fail if these certificates were updated by deleting the signing key. This would cause the console to not recover when CA certs were rotated. With this release, console conversion webhook server was updated to detect CA certificate changes, and handle them at runtime. The server now remains available and the console recovers as expected after CA certificates are rotated. (link:https://issues.redhat.com/browse/OCPBUGS-15827[*OCPBUGS-15827*])

> * Previously, production builds of the console front-end bundle have historically had source maps disabled. This meant browser tools for analyzing source code could not be used on production builds. With this release, the console Webpack configuration is updated to enable source maps on production builds. Browser tools will now work as expected for both dev and production builds. (link:https://issues.redhat.com/browse/OCPBUGS-10851[*OCPBUGS-10851*])

Reviewer: ```suggestion
* Previously, production builds of the console front-end bundle have historically had source maps disabled. As a consequence, browser tools for analyzing source code could not be used on production builds. With this release, the console Webpack configuration is updated to enable source maps on production builds. Browser tools will now work as expected for both dev and production builds. (link:https://issues.redhat.com/browse/OCPBUGS-10851[*OCPBUGS-10851*])
```
